internal/gitserver/gitdomain/exec.go

```diff
@@ -128,8 +128,34 @@ func isAllowedGitArg(allowedArgs []string, arg string) bool {
 	return false
 }
 
+// isAllowedDiffPathArg checks if the diff path arg is allowed.
+func isAllowedDiffPathArg(arg string, repoDir string) bool {
+	// allows diff command path that requires (dot) as path
+	// example: diff --find-renames ... --no-prefix commit -- .
+	if arg == "." {
+		return true
+	}
+
+	arg = filepath.Clean(arg)
+	if !filepath.IsAbs(arg) {
+		arg = filepath.Join(repoDir, arg)
+	}
+	filePath, err := filepath.Abs(arg)
+
+	if err != nil {
+		return false
+	} else {
+		// Check if absolute path is a sub path of the repo dir
+		repoRoot, err := filepath.Abs(repoDir)
+		if err == nil {
+			return strings.HasPrefix(filePath, repoRoot)
+		}
+	}
+	return false
```

Comment:
this feels easier to read to me, wdyt? the else branch after a return is not necessary, for example :)

```suggestion
	arg = filepath.Clean(arg)
	if !filepath.IsAbs(arg) {
		arg = filepath.Join(repoDir, arg)
	}
	filePath, err := filepath.Abs(arg)
	if err != nil {
		return false
	}
	
		// Check if absolute path is a sub path of the repo dir
		repoRoot, err := filepath.Abs(repoDir)
		if err != nil {
		return false
		}
			return strings.HasPrefix(filePath, repoRoot)
```